src/foam/core/AbstractFObjectPropertyInfo.java

```diff
@@ -105,7 +106,7 @@ public boolean hardDiff(FObject o1, FObject o2, FObject diff) {
      *  X.get("DAO").put(obj);
      * In this case: the before value inside the model is loss, so can not find difference
      */
-    if ( (this.get(o1) == this.get(o2)) ) {
+    if ( SafetyUtil.compare(o1, o2) == 0 ) {
```

Comment:
== wasn't correct because it wouldn't compare Strings correctly

Reply:
This should be SafetyUtil.compare(this.get(o1), this.get(o2))  == 0 or SafetyUtil.equals(this.get(o1), this.get(o2)) or just compare(o1, o2), since compare is a method of PropertyInfo and the implementation is probably something like:
      public int compare(Object o1, Object o2) {
        return foam.util.SafetyUtil.compare(get_(o1), get_(o2));
      }